chipsec/hal/paging.py

```diff
@@ -199,73 +203,67 @@ def read_entries(self, info, addr, size=8):
 
-    def print_info(self, name):
-        logger().log('\n  {} physical address ranges:'.format(name))
+    def print_info(self, name: str) -> None:
+        logger().log(f'\n  {name} physical address ranges:')
         mem_range = self.get_mem_range()
         for index in range(len(mem_range)):
             i = mem_range[index]
-            logger().log('    0x{:013X} - 0x{:013X} {:8d}  {}'.format(i[0], i[1] - 1, (i[1] - i[0]) >> 12, i[2]))
+            logger().log(f'    0x{i[0]:013X} - 0x{i[1] - 1:013X} {(i[1] - i[0]) >> 12:8d}  {i[2]}')
 
-        logger().log('\n  {} pages:'.format(name))
+        logger().log(f'\n  {name} pages:')
         for i in sorted(self.pt.keys()):
-            logger().log('    0x{:013X}  {}'.format(i, self.pt[i]))
+            logger().log(f'    0x{i:013X}  {self.pt[i]}')
         logger().log('\n')
-        logger().log('  {} size: {:d} KB, address space: {:d} MB'.format(name, len(self.pt.keys()) * 4, self.get_address_space() >> 20))
+        logger().log(f'  {name} size: {len(self.pt.keys()) * 4:d} KB, address space: {self.get_address_space() >> 20:d} MB')
         return
 
-    def check_misconfig(self, addr_list):
+    def check_misconfig(self, addr_list: List[int]) -> None:
         addr_list = [x & MAXPHYADDR for x in addr_list]
         mem_range = self.get_mem_range()
         for addr in addr_list:
             for i in range(len(mem_range)):
                 if (mem_range[i][0] <= addr) and (addr < mem_range[i][1]):
-                    print('*** WARNING: PAGE TABLES MISCONFIGURATION  0x{:013X}'.format(addr))
+                    logger().log_hal(f'*** WARNING: PAGE TABLES MISCONFIGURATION  0x{addr:013X}')
         return
 
-    def save_configuration(self, path):
-        cfg = open(path, 'w')
-        try:
-            cfg.write(str(self.translation_level2.translation) + '\n')
-            cfg.write(str(self.translation) + '\n')
-            cfg.write(str(self.pt))
-        finally:
-            cfg.close()
+    def save_configuration(self, path: str) -> None:
+        with open(path, 'w') as cfg:
+            try:
+                cfg.write(str(self.translation_level2.translation) + '\n')
+                cfg.write(str(self.translation) + '\n')
+                cfg.write(str(self.pt))
+            except:
+                logger().log_hal(f'[paging] Error saving: {path}')
         return
 
-    def load_configuration(self, path):
-        cfg = open(path, 'r')
-        try:
-            self.translation_level2.translation = eval(cfg.readline())
-            self.translation = eval(cfg.readline())
-            self.pt = eval(cfg.readline())
-        finally:
-            cfg.close()
+    def load_configuration(self, path: str) -> None:
+        with open(path, 'r') as cfg:
+            try:
+                self.translation_level2.translation = eval(cfg.readline())
+                self.translation = eval(cfg.readline())
+                self.pt = eval(cfg.readline())
+            finally:
```

Comment:
This is a `finally` clause, not an `except` clause. Did you want to change it to an `except`?